apps/dashboard/src/lib/components/Course/components/Settings/index.svelte

```diff
@@ -169,11 +175,16 @@
         grading: !!course.metadata.grading,
         lesson_download: !!course.metadata.lessonDownload,
         is_published: !!course.is_published,
-        allow_new_students: course.metadata.allowNewStudent
+        allow_new_students: course.metadata.allowNewStudent,
       };
     }
   }
 
+  const generateNewCourseLink = async () => {
+    newSlug = await generateSlug($course.title);
```

Comment:
I think a couple of things here:

1. I don't think `newSlug` is necessary, you can just update $course.slug directly.
2. `generateSlug` is not an async function, which means you don't need to add `await`.


```js
const generateNewCourseLink = () => {
  $course.slug = generateSlug($course.title);
}
```

And then everywhere you have newSlug just replace it to `$course.slug`

Reply:
@R4Rohit23, heads up for next time. Please don't resolve a comment made by the reviewer. It is the responsibility of the reviewer to resolve the comment if he considers it resolved.

Thanks